### ISSUE TITLE
Legacy block-comment purpose extraction does not skip PHP open tags

### ISSUE THREAD

**Issue by styler-ai** [styler-ai/ProjectAtlas#263]
## Bug

GitHub Codex flagged this in the Python-era implementation on PR #104. The source file is gone, but the same edge case still exists in the Rust legacy purpose-header extractor when a project configures `.php` as `block-comment`.

Original Codex discussion: https://github.com/styler-ai/ProjectAtlas/pull/104#discussion_r2659145008

## Current Evidence

`crates/projectatlas-cli/src/atlas_map.rs::extract_block_comment_purpose` finds the first content line and requires it to start with `/*`. A normal PHP file commonly starts with `<?php` followed by a block Purpose header:

```php
<?php
/* Purpose: Legacy PHP module. */
```

That layout is still reported as missing a block-comment Purpose header if `.php` is configured to use `block-comment` legacy import style.

## Expected

Legacy Purpose header import should skip an initial PHP opening tag before checking for a block comment header.

## Suggested Architecture

- Teach `extract_block_comment_purpose` or `resolve_purpose_style` handling to skip a leading `<?php` line for PHP files.
- Add a regression with `.php = "block-comment"` and a `<?php` opener before the Purpose block.
- Keep ProjectAtlas 3's current rule intact: source Purpose headers are migration input only, not required durable metadata.

## Acceptance Criteria

- PHP files with an opener followed by a block Purpose header import the purpose correctly when configured for block-comment style.
- Non-PHP block-comment behavior remains unchanged.
- Docs continue to describe source headers as legacy migration input only.
